cmd/ocm/create/machinepool/cmd.go

```diff
@@ -80,84 +84,136 @@ func init() {
 	Cmd.MarkFlagRequired("instance-type")
 
 	flags.IntVar(
-		&args.replicas,
+		&args.Replicas,
 		"replicas",
 		0,
 		"Count of machines for this machine pool.",
 	)
 
-	arguments.AddAutoscalingFlags(flags, &args.autoscaling)
+	arguments.AddAutoscalingFlags(flags, &args.Autoscaling)
 
 	flags.StringVar(
-		&args.labels,
+		&args.Labels,
 		"labels",
 		"",
 		"Labels for machine pool. Format should be a comma-separated list of 'key=value'. "+
 			"This list will overwrite any modifications made to Node labels on an ongoing basis.",
 	)
 
 	flags.StringVar(
-		&args.taints,
+		&args.Taints,
 		"taints",
 		"",
 		"Taints for machine pool. Format should be a comma-separated list of 'key=value:scheduleType'. "+
 			"This list will overwrite any modifications made to Node taints on an ongoing basis.",
 	)
 
-	flags.StringSliceVar(&args.additionalSecurityGroupIds,
+	flags.StringSliceVar(&args.AdditionalSecurityGroupIds,
 		additionalSecurityGroupIdsFlag,
 		nil,
 		"The additional Security Group IDs to be added to the machine pool. "+
 			"Format should be a comma-separated list.",
 	)
+
+	flags.StringVar(
+		&args.AvailabilityZone,
+		"availability-zone",
+		"",
+		"Select availability zone to create a single AZ machine pool for a multi-AZ cluster",
+	)
 }
 
 func run(cmd *cobra.Command, argv []string) error {
+	args.Argv = argv
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return fmt.Errorf("Failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	cluster, err := getCluster(connection, args.ClusterKey)
+	if err != nil {
+		return err
+	}
+
+	if err := VerifyCluster(cluster); err != nil {
+		return err
+	}
+
+	if err := VerifyArguments(
+		args,
+		&flagSet{cmd.Flags()},
+		&machineTypeListGetter{connection},
+		cluster,
+	); err != nil {
+		return err
+	}
+
+	machinePoolId := argv[0]
+
+	machinePool, err := buildMachinePool(machinePoolId)
+	if err != nil {
+		return err
+	}
 
-	// Check that the cluster key (name, identifier or external identifier) given by the user
-	// is reasonably safe so that there is no risk of SQL injection:
-	clusterKey := args.clusterKey
-	if !c.IsValidClusterKey(clusterKey) {
-		return fmt.Errorf(
-			"Cluster name, identifier or external identifier '%s' isn't valid: it "+
-				"must contain only letters, digits, dashes and underscores",
-			clusterKey,
-		)
+	if err := addMachinePoolToCluster(connection, cluster.Id(), machinePool); err != nil {
+		return err
 	}
 
-	if len(argv) < 1 || argv[0] == "" {
+	fmt.Printf("Machine pool '%s' created on cluster '%s'\n", machinePoolId, args.ClusterKey)
+	return nil
+}
+
+func getCluster(
+	connection *sdk.Connection,
+	clusterKey string,
+) (ocm.Cluster, error) {
+	clusterData, err := c.GetCluster(connection, clusterKey)
```

Comment:
Looking like the check to verify clusterKey was removed and there is no other check done to see if clusterKey value was input at all by the user (here or in the calling function "run") before try to fetch the cluster from the backend.

There  should be a check to determine if that option is atleast set and if not, user should be prompted to provide a value rather than going to the backend,  You would still get an error from the backend but that call firstly isn't needed and the error msg prompted to the user wouldn't be correct either.